File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    var answer = [];
    let [a,b] = gcd(n,m);
    return [a,a*b];
}

function gcd(n,m){
    let num = 0;
    let least = 0;
    
    for(let i = 0 ; i < Math.max(n,m); i++){
        if(n % i == 0 && m % i == 0){
            num = i;
        }
    }
    least = (n / num) * (m / num);

    return [num,least];
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.util.Arrays;

public class Solution {
    public int[] solution(int n, int m) {
        int[] answer = {gcd(n,m),lcm(n,m)};

        return answer;
    }

    public static int gcd(int a, int b){
        int max = Math.max(a,b);
        int min = Math.min(a,b);

        if(min == 0) return max;

        return gcd(min, max % min);
    }

    public static int lcm(int a, int b){
        return a*b / gcd(a,b);
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int n = 3; int m = 12;
        System.out.println(Arrays.toString(sol.solution(n,m)));
    }
}
